### ISSUE TITLE
Soldaten laufen ineinander

### ISSUE THREAD

**Issue by Spikeone** [Return-To-The-Roots/s25client#84]
Nach dem Angriff auf den Hafen, liefen die Soldaten zurück, während 2 direkt übereinander gelaufen sind, im originalen sind allerdings soldaten nie durch andere hindurch gelaufen.
Zu sehen bei dem Angriff von weiß auf Blau

Spike (fvv1) wrote on 2010-09-11:	#1
2010-09-11_15-23-35.rpl Edit (15.0 KiB, application/octet-stream)

https://bugs.launchpad.net/s25rttr/+bug/635753

YES, this bug IS still valid.

**Comment by Flamefire** [comment]
I don't think we have collision detection somewhere. Every figure just ignores other figures and situations like this are usually avoided by randomizing the walk direction.

Note that fixing this may significantly increase the time it takes for soldiers returning because the path back might be the same for all soldiers so they will queue up on a single point...

**Comment by Spikeone** [comment]
Well - it'd be quite simple (at least in my mind): If a soldiers moves to a node, that node is occupied. A node that is occupied can't be used by an other soldier for it's pathfinding. When a node is reached, path gets recalculated. That way it should work like it did in the original and would only take some more time.

**Comment by Spikeone** [comment]
![image](https://user-images.githubusercontent.com/1276342/50424351-b2bc1a80-0862-11e9-9491-2dc026c72d00.png)

**Comment by Flamefire** [comment]
Possible problem: Returning soldiers could block attacking soldiers leading to a deadlock. Solution: returning soldiers do not block

**Comment by Spikeone** [comment]
While playing around a bit - soldiers actually walked inside each other when changing occupation for a border group.
